File: 171.excel-sheet-column-number.go
/*
 * @lc app=leetcode id=171 lang=golang
 *
 * [171] Excel Sheet Column Number
 */

package golang

// @lc code=start
func titleToNumber(s string) int {
	col := 0
	for i := 0; i < len(s); i++ {
		c := int(s[i]-'A') + 1
		col *= 26
		col += c
	}
	return col
}

// @lc code=end

File: typescript/src/171.excel-sheet-column-number.ts
/*
 * @lc app=leetcode id=171 lang=typescript
 *
 * [171] Excel Sheet Column Number
 */

// @lc code=start
function titleToNumber(s: string): number {
    let num = 0
    const a = "A".charCodeAt(0) - 1
    for (let i = 0; i < s.length; i++) {
        const element = s.charCodeAt(i);
        num *= 26
        num += (element - a)
    }
    return num
};
// @lc code=end

